tests/Client/TestRepo.php

```diff
@@ -25,6 +25,8 @@ class TestRepo implements RepoFileFetcherInterface
      */
     public $fileContents = [];
 
+    public array $fetchMetadataArguments = [];
```

Comment:
This really not clear what this is for. 

I am a kinda of understand because I looking at the specific PR where you are adidng. but if I came back in a year and was looking at the codebase in another context and stumble upon this  I think I would throughly confused.

Reply:
so really for this and anywhere it is used but generally, the declaration, the setting and reading have a relationship and reasoning but it isn't explained

Reply:
Agreed. I added a comment.